Update etmv4.c

jeremyng123 opened the pull request:
in ihi0064F etm4.0-etm4.4 architecture spectification (pg 6-282), Atom Format 6 Instruction trace packet only permits count values from 0b00000 to 0b10100 (0 to 20)

hwangcc23 commented on etmv4.c:
```diff
@@ -1066,6 +1066,12 @@ DECL_DECODE_FN(atom_format_6)
     COUNT = pkt[0] & 0x1f;
+    
+    if (COUNT > 20) {
+        LOGE("Invalid COUNT in a ATOM format 6 packet\n");
```
Typo here
"an ATOM"

hwangcc23 commented on etmv4.c:
```diff
@@ -1066,6 +1066,12 @@ DECL_DECODE_FN(atom_format_6)
+    
+    if (COUNT > 20) {
+        LOGE("Invalid COUNT in a ATOM format 6 packet\n");
+        return 1;
```
Should return -1 to indicate the error.

hwangcc23 commented on etmv4.c:
```diff
@@ -59,6 +59,7 @@ DEF_TRACEPKT(address_context_32bit_is0, 0xff, 0x82);
 DEF_TRACEPKT(address_context_32bit_is1, 0xff, 0x83);
 DEF_TRACEPKT(address_context_64bit_is0, 0xff, 0x85);
 DEF_TRACEPKT(address_context_64bit_is1, 0xff, 0x86);
+// changes masks values for atom_format_1-4: jeremyng123
```
This change (to add the comment line) is not needed.

hwangcc23 reviewed (changes requested):
Please help check comments, squash them, and resend the PR.
Thanks.

jeremyng123 commented:
I have resolved your comments. Please verify the changes :)

hwangcc23 commented:
OK for me.
Please squash all commits into one commit. Thanks.
Reference: https://github.com/wprig/wprig/wiki/How-to-squash-commits

Commit:
Changes:
1. Fixed atom mask values to better represent etm's atom elements
2. Fixed permitted range values for atom_format_5. Count values only permit 0 to 20.

Jeremy Ng <ngyzj95@gmail.com>

jeremyng123 commented:
Hi, I didn't exactly squash the commits because there were some issues. but i have redid the PR to only have 1 commit. there shouldn't be any difference in the changes as discussed.

hwangcc23 commented:
Some of your changes (fix-typo) are not squashed into your final patch. To avoid micro-management, I will merge your patch directly, and fix other problems later directly.
Thanks for your contribution.
